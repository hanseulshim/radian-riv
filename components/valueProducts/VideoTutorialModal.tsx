import React from 'react'
import Modal from 'components/common/Modal'

interface Props {
  closeModal: () => void
}

const data = [
  {
    title: 'Search Screen',
    fileName:
      'https://s3.amazonaws.com/assets.boostlabs/radian-riv/search-screen.mp4',
    description: 'How to Search for PReviously Placed Orders',
    duration: '4:32',
    updated: '03/07/2018'
  },
  {
    title: 'How to Place a Single Order',
    fileName:
      'https://s3.amazonaws.com/assets.boostlabs/radian-riv/single-order.mp4',
    description: 'How to place a single Radian Interactive Value order.',
    duration: '3:29',
    updated: '03/07/2018'
  },
  {
    title: 'How to Place a Bulk Order',
    fileName:
      'https://s3.amazonaws.com/assets.boostlabs/radian-riv/bulk-order.mp4',
    description:
      'Place More Than one Radian Interactive Value Order at Once in Red Bell',
    duration: '3:40',
    updated: '03/07/2018'
  },
  {
    title: 'Subject Property Information',
    fileName:
      'https://s3.amazonaws.com/assets.boostlabs/radian-riv/subject-property.mp4',
    description: 'Where to Find Information about the Subject Property',
    duration: '2:57',
    updated: '03/07/2018'
  },
  {
    title: 'Comparable Properties',
    fileName:
      'https://s3.amazonaws.com/assets.boostlabs/radian-riv/comparable-properties.mp4',
    description: 'Information on Comparable Properties',
    duration: '9:10',
    updated: '03/07/2018'
  },
  {
    title: 'Define Market Area',
    fileName:
      'https://s3.amazonaws.com/assets.boostlabs/radian-riv/market-area.mp4',
    description: 'How to Filter Comparable Properties',
    duration: '5:13',
    updated: '03/07/2018'
  },
  {
    title: 'How to Load Manual Comparables',
    fileName:
      'https://s3.amazonaws.com/assets.boostlabs/radian-riv/manual-comparables.mp4',
    description:
      'How to Load Manual Comparables into the Radian Interactive Value product.',
    duration: '6:48',
    updated: '10/04/2018'
  },
  {
    title: 'Listing Sheets',
    fileName:
      'https://s3.amazonaws.com/assets.boostlabs/radian-riv/listing-sheets.mp4',
    description: 'How to find and use the Red Bell listing sheets.',
    duration: '7:39',
    updated: '10/04/2018'
  },
  {
    title: 'Changing Subject Property Characteristics',
    fileName:
      'https://s3.amazonaws.com/assets.boostlabs/radian-riv/subject-property-characteristics.mp4',
    description:
      'How to Change the Subject Property Characteristics in the Radian Interactive Value tool.',
    duration: '10:53',
    updated: '10/04/2018'
  },
  {
    title: 'Flip Report',
    fileName:
      'https://s3.amazonaws.com/assets.boostlabs/radian-riv/flip-report.mp4',
    description: 'How to see properties that have flipped in an area.',
    duration: '10:53',
    updated: '10/04/2018'
  }
]

export default function VideoTutorialModal({ closeModal }: Props) {
  return (
    <Modal closeModal={closeModal} title={''} width={1360} id="video-modal">
      <h2>Video Tutorials</h2>
      <div className="video-container">
        {data.map(videoObj => (
          <div className="video" key={videoObj.title}>
            <div className="title">{videoObj.title}</div>
            <video className="video-file" controls controlsList="nodownload">
              <source src={videoObj.fileName} type="video/mp4" />
            </video>
            <div className="description-container">
              <div className="description">{videoObj.description}</div>
              <div className="video-stats">
                <div>Duration {videoObj.duration}</div>
                <div>{videoObj.updated}</div>
              </div>
            </div>
          </div>
        ))}
      </div>
    </Modal>
  )
}
