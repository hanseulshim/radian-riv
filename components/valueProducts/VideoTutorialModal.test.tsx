import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import VideoTutorialModal from './VideoTutorialModal'

describe('VideoTutorialModal', () => {
  let component
  beforeAll(() => {
    const props = {
      closeModal: () => {}
    }
    component = shallow(<VideoTutorialModal {...props} />)
  })
  it('has video-modal container', () => {
    expect(component.find('#video-modal')).toHaveLength(1)
    expect(component.find('h2').text()).toEqual('Video Tutorials')
    expect(component.find('.video-container')).toHaveLength(1)
  })
  it('has list of video containers content', () => {
    expect(component.find('.video')).toHaveLength(10)
    expect(component.find('.title')).toHaveLength(10)
    expect(component.find('.video-file')).toHaveLength(10)
    expect(component.find('.description-container')).toHaveLength(10)
    expect(component.find('.description')).toHaveLength(10)
    expect(component.find('.video-stats')).toHaveLength(10)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
