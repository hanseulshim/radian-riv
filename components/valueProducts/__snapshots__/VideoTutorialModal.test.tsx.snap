// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoTutorialModal should match snapshot 1`] = `
<Modal
  closeModal={[Function]}
  id="video-modal"
  title=""
  width={1360}
>
  <h2>
    Video Tutorials
  </h2>
  <div
    className="video-container"
  >
    <div
      className="video"
      key="Search Screen"
    >
      <div
        className="title"
      >
        Search Screen
      </div>
      <video
        className="video-file"
        controls={true}
        controlsList="nodownload"
      >
        <source
          src="https://s3.amazonaws.com/assets.boostlabs/radian-riv/search-screen.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="description-container"
      >
        <div
          className="description"
        >
          How to Search for PReviously Placed Orders
        </div>
        <div
          className="video-stats"
        >
          <div>
            Duration 
            4:32
          </div>
          <div>
            03/07/2018
          </div>
        </div>
      </div>
    </div>
    <div
      className="video"
      key="How to Place a Single Order"
    >
      <div
        className="title"
      >
        How to Place a Single Order
      </div>
      <video
        className="video-file"
        controls={true}
        controlsList="nodownload"
      >
        <source
          src="https://s3.amazonaws.com/assets.boostlabs/radian-riv/single-order.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="description-container"
      >
        <div
          className="description"
        >
          How to place a single Radian Interactive Value order.
        </div>
        <div
          className="video-stats"
        >
          <div>
            Duration 
            3:29
          </div>
          <div>
            03/07/2018
          </div>
        </div>
      </div>
    </div>
    <div
      className="video"
      key="How to Place a Bulk Order"
    >
      <div
        className="title"
      >
        How to Place a Bulk Order
      </div>
      <video
        className="video-file"
        controls={true}
        controlsList="nodownload"
      >
        <source
          src="https://s3.amazonaws.com/assets.boostlabs/radian-riv/bulk-order.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="description-container"
      >
        <div
          className="description"
        >
          Place More Than one Radian Interactive Value Order at Once in Red Bell
        </div>
        <div
          className="video-stats"
        >
          <div>
            Duration 
            3:40
          </div>
          <div>
            03/07/2018
          </div>
        </div>
      </div>
    </div>
    <div
      className="video"
      key="Subject Property Information"
    >
      <div
        className="title"
      >
        Subject Property Information
      </div>
      <video
        className="video-file"
        controls={true}
        controlsList="nodownload"
      >
        <source
          src="https://s3.amazonaws.com/assets.boostlabs/radian-riv/subject-property.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="description-container"
      >
        <div
          className="description"
        >
          Where to Find Information about the Subject Property
        </div>
        <div
          className="video-stats"
        >
          <div>
            Duration 
            2:57
          </div>
          <div>
            03/07/2018
          </div>
        </div>
      </div>
    </div>
    <div
      className="video"
      key="Comparable Properties"
    >
      <div
        className="title"
      >
        Comparable Properties
      </div>
      <video
        className="video-file"
        controls={true}
        controlsList="nodownload"
      >
        <source
          src="https://s3.amazonaws.com/assets.boostlabs/radian-riv/comparable-properties.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="description-container"
      >
        <div
          className="description"
        >
          Information on Comparable Properties
        </div>
        <div
          className="video-stats"
        >
          <div>
            Duration 
            9:10
          </div>
          <div>
            03/07/2018
          </div>
        </div>
      </div>
    </div>
    <div
      className="video"
      key="Define Market Area"
    >
      <div
        className="title"
      >
        Define Market Area
      </div>
      <video
        className="video-file"
        controls={true}
        controlsList="nodownload"
      >
        <source
          src="https://s3.amazonaws.com/assets.boostlabs/radian-riv/market-area.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="description-container"
      >
        <div
          className="description"
        >
          How to Filter Comparable Properties
        </div>
        <div
          className="video-stats"
        >
          <div>
            Duration 
            5:13
          </div>
          <div>
            03/07/2018
          </div>
        </div>
      </div>
    </div>
    <div
      className="video"
      key="How to Load Manual Comparables"
    >
      <div
        className="title"
      >
        How to Load Manual Comparables
      </div>
      <video
        className="video-file"
        controls={true}
        controlsList="nodownload"
      >
        <source
          src="https://s3.amazonaws.com/assets.boostlabs/radian-riv/manual-comparables.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="description-container"
      >
        <div
          className="description"
        >
          How to Load Manual Comparables into the Radian Interactive Value product.
        </div>
        <div
          className="video-stats"
        >
          <div>
            Duration 
            6:48
          </div>
          <div>
            10/04/2018
          </div>
        </div>
      </div>
    </div>
    <div
      className="video"
      key="Listing Sheets"
    >
      <div
        className="title"
      >
        Listing Sheets
      </div>
      <video
        className="video-file"
        controls={true}
        controlsList="nodownload"
      >
        <source
          src="https://s3.amazonaws.com/assets.boostlabs/radian-riv/listing-sheets.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="description-container"
      >
        <div
          className="description"
        >
          How to find and use the Red Bell listing sheets.
        </div>
        <div
          className="video-stats"
        >
          <div>
            Duration 
            7:39
          </div>
          <div>
            10/04/2018
          </div>
        </div>
      </div>
    </div>
    <div
      className="video"
      key="Changing Subject Property Characteristics"
    >
      <div
        className="title"
      >
        Changing Subject Property Characteristics
      </div>
      <video
        className="video-file"
        controls={true}
        controlsList="nodownload"
      >
        <source
          src="https://s3.amazonaws.com/assets.boostlabs/radian-riv/subject-property-characteristics.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="description-container"
      >
        <div
          className="description"
        >
          How to Change the Subject Property Characteristics in the Radian Interactive Value tool.
        </div>
        <div
          className="video-stats"
        >
          <div>
            Duration 
            10:53
          </div>
          <div>
            10/04/2018
          </div>
        </div>
      </div>
    </div>
    <div
      className="video"
      key="Flip Report"
    >
      <div
        className="title"
      >
        Flip Report
      </div>
      <video
        className="video-file"
        controls={true}
        controlsList="nodownload"
      >
        <source
          src="https://s3.amazonaws.com/assets.boostlabs/radian-riv/flip-report.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="description-container"
      >
        <div
          className="description"
        >
          How to see properties that have flipped in an area.
        </div>
        <div
          className="video-stats"
        >
          <div>
            Duration 
            10:53
          </div>
          <div>
            10/04/2018
          </div>
        </div>
      </div>
    </div>
  </div>
</Modal>
`;
